netbox/dcim/api/views.py

```diff
@@ -672,6 +673,12 @@ def list(self, request):
         peer_device_name = request.query_params.get(self._device_param.name)
         peer_interface_name = request.query_params.get(self._interface_param.name)
 
+        # check permissions of user, if login is required
+        if settings.LOGIN_REQUIRED:
+            if not request.user.has_perm('dcim.view_device') or not request.user.has_perm('dcim.view_interface'):
+                raise PermissionDenied('You do not have permission to perform this action.')
```

Comment:
No need to specify an error message when raising `PermissionDenied`. It includes a suitable default message.

Reply:
Ok, thanks. I didn't notice that.